diff --git a/pidgin/doc/pidgin.1.in b/pidgin/doc/pidgin.1.in
index b64fd47..8c0eaac 100644
--- a/pidgin/doc/pidgin.1.in
+++ b/pidgin/doc/pidgin.1.in
@@ -1,4 +1,4 @@
-Ri.\" Copyright (c) 2000, Dennis Ristuccia <dennis@dennisr.net>
+.\" Copyright (c) 2000, Dennis Ristuccia <dennis@dennisr.net>
 .\"
 .\" This is free documentation; you can redistribute it and/or
 .\" modify it under the terms of the GNU General Public License as
