diff -ur pidgin-2.4.1.orig/libpurple/protocols/oscar/oscar.c pidgin-2.4.1/libpurple/protocols/oscar/oscar.c
--- pidgin-2.4.1.orig/libpurple/protocols/oscar/oscar.c	2008-03-31 20:21:24 +0300
+++ pidgin-2.4.1/libpurple/protocols/oscar/oscar.c	2008-04-30 23:56:09 +0300
@@ -3823,8 +3823,10 @@
 				utf8 && *utf8 ? utf8 : "");
 		g_free(utf8);
 
-		oscar_user_info_convert_and_add(account,
-				user_info, _("Status"), tmp);
+//		oscar_user_info_convert_and_add(account,
+//				user_info, _("Status"), tmp);
+		purple_notify_user_info_add_pair(user_info, _("Status"), purple_status_get_name(status));
+
 	}
 
 	oscar_user_info_convert_and_add(account, user_info, _("Additional Information"), info->info);
