diff -urN pidgin-2.1.0.orig/libpurple/plugins/Makefile.am pidgin-2.1.0/libpurple/plugins/Makefile.am
--- pidgin-2.1.0.orig/libpurple/plugins/Makefile.am	2007-05-10 06:02:58 +0300
+++ pidgin-2.1.0/libpurple/plugins/Makefile.am	2007-08-18 16:20:41 +0300
@@ -78,27 +78,27 @@
 simple_la_SOURCES			= simple.c
 statenotify_la_SOURCES      = statenotify.c
 
-autoaccept_la_LIBADD        = $(GLIB_LIBS)
-buddynote_la_LIBADD         = $(GLIB_LIBS)
-ciphertest_la_LIBADD		= $(GLIB_LIBS)
-codeinline_la_LIBADD		= $(GLIB_LIBS)
-idle_la_LIBADD              = $(GLIB_LIBS)
-joinpart_la_LIBADD          = $(GLIB_LIBS)
-log_reader_la_LIBADD        = $(GLIB_LIBS)
-newline_la_LIBADD           = $(GLIB_LIBS)
-offlinemsg_la_LIBADD        = $(GLIB_LIBS)
-pluginpref_example_la_LIBADD = $(GLIB_LIBS)
-psychic_la_LIBADD           = $(GLIB_LIBS)
-signals_test_la_LIBADD		= $(GLIB_LIBS)
-simple_la_LIBADD			= $(GLIB_LIBS)
-statenotify_la_LIBADD       = $(GLIB_LIBS)
+autoaccept_la_LIBADD        = $(GLIB_LIBS) ../libpurple.la
+buddynote_la_LIBADD         = $(GLIB_LIBS) ../libpurple.la
+ciphertest_la_LIBADD		= $(GLIB_LIBS) ../libpurple.la
+codeinline_la_LIBADD		= $(GLIB_LIBS) ../libpurple.la
+idle_la_LIBADD              = $(GLIB_LIBS) ../libpurple.la
+joinpart_la_LIBADD          = $(GLIB_LIBS) ../libpurple.la
+log_reader_la_LIBADD        = $(GLIB_LIBS) ../libpurple.la
+newline_la_LIBADD           = $(GLIB_LIBS) ../libpurple.la
+offlinemsg_la_LIBADD        = $(GLIB_LIBS) ../libpurple.la
+pluginpref_example_la_LIBADD = $(GLIB_LIBS) ../libpurple.la
+psychic_la_LIBADD           = $(GLIB_LIBS) ../libpurple.la
+signals_test_la_LIBADD		= $(GLIB_LIBS) ../libpurple.la
+simple_la_LIBADD			= $(GLIB_LIBS) ../libpurple.la
+statenotify_la_LIBADD       = $(GLIB_LIBS) ../libpurple.la
 
 if ENABLE_DBUS
 
 CLEANFILES              = dbus-example-bindings.c
 dbus_example_la_SOURCES = dbus-example.c
 
-dbus_example_la_LIBADD      = $(GLIB_LIBS) $(DBUS_LIBS)
+dbus_example_la_LIBADD      = $(GLIB_LIBS) $(DBUS_LIBS) ../libpurple.la
 
 .PHONY: always
 
diff -urN pidgin-2.1.0.orig/libpurple/plugins/ssl/Makefile.am pidgin-2.1.0/libpurple/plugins/ssl/Makefile.am
--- pidgin-2.1.0.orig/libpurple/plugins/ssl/Makefile.am	2007-05-03 21:14:10 +0300
+++ pidgin-2.1.0/libpurple/plugins/ssl/Makefile.am	2007-08-18 16:46:46 +0300
@@ -18,9 +18,9 @@
 ssl_gnutls_la_SOURCES = ssl-gnutls.c
 ssl_nss_la_SOURCES    = ssl-nss.c
 
-ssl_la_LIBADD        = $(GLIB_LIBS)
-ssl_gnutls_la_LIBADD = $(GLIB_LIBS) $(GNUTLS_LIBS)
-ssl_nss_la_LIBADD    = $(GLIB_LIBS) $(NSS_LIBS)
+ssl_la_LIBADD        = $(GLIB_LIBS) ../../libpurple.la
+ssl_gnutls_la_LIBADD = $(GLIB_LIBS) $(GNUTLS_LIBS) ../../libpurple.la
+ssl_nss_la_LIBADD    = $(GLIB_LIBS) $(NSS_LIBS) ../../libpurple.la
 
 endif # PLUGINS
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/gg/Makefile.am pidgin-2.1.0/libpurple/protocols/gg/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/gg/Makefile.am	2007-05-03 21:14:10 +0300
+++ pidgin-2.1.0/libpurple/protocols/gg/Makefile.am	2007-08-18 16:23:23 +0300
@@ -57,7 +57,7 @@
 
 libgg_a_SOURCES = $(GGSOURCES)
 libgg_a_CFLAGS  = $(AM_CFLAGS)
-libgg_a_LIBADD  = $(GADU_LIBS)
+libgg_a_LIBADD  = $(GADU_LIBS) ../../libpurple.la
 
 else
 
@@ -66,7 +66,7 @@
 noinst_LIBRARIES =
 
 libgg_la_SOURCES = $(GGSOURCES)
-libgg_la_LIBADD  = $(GLIB_LIBS) $(GADU_LIBS)
+libgg_la_LIBADD  = $(GLIB_LIBS) $(GADU_LIBS) ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/irc/Makefile.am pidgin-2.1.0/libpurple/protocols/irc/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/irc/Makefile.am	2007-05-10 06:02:58 +0300
+++ pidgin-2.1.0/libpurple/protocols/irc/Makefile.am	2007-08-18 16:23:37 +0300
@@ -24,7 +24,7 @@
 noinst_LIBRARIES =
 
 libirc_la_SOURCES = $(IRCSOURCES)
-libirc_la_LIBADD  = $(GLIB_LIBS)
+libirc_la_LIBADD  = $(GLIB_LIBS) ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/jabber/Makefile.am pidgin-2.1.0/libpurple/protocols/jabber/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/jabber/Makefile.am	2007-06-12 09:13:03 +0300
+++ pidgin-2.1.0/libpurple/protocols/jabber/Makefile.am	2007-08-18 16:30:22 +0300
@@ -56,10 +56,10 @@
 noinst_LIBRARIES =
 
 libjabber_la_SOURCES = $(JABBERSOURCES)
-libjabber_la_LIBADD = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS)
+libjabber_la_LIBADD = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS) ../../libpurple.la
 
 libxmpp_la_SOURCES = libxmpp.c
-libxmpp_la_LIBADD = libjabber.la
+libxmpp_la_LIBADD = libjabber.la ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/msn/Makefile.am pidgin-2.1.0/libpurple/protocols/msn/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/msn/Makefile.am	2007-05-03 21:14:10 +0300
+++ pidgin-2.1.0/libpurple/protocols/msn/Makefile.am	2007-08-18 16:31:57 +0300
@@ -79,7 +79,7 @@
 st =
 pkg_LTLIBRARIES   = libmsn.la
 libmsn_la_SOURCES = $(MSNSOURCES)
-libmsn_la_LIBADD  = $(GLIB_LIBS)
+libmsn_la_LIBADD  = $(GLIB_LIBS) ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/novell/Makefile.am pidgin-2.1.0/libpurple/protocols/novell/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/novell/Makefile.am	2007-05-03 21:14:10 +0300
+++ pidgin-2.1.0/libpurple/protocols/novell/Makefile.am	2007-08-18 16:33:34 +0300
@@ -46,7 +46,7 @@
 noinst_LIBRARIES =
 
 libnovell_la_SOURCES = $(NOVELLSOURCES)
-libnovell_la_LIBADD  = $(GLIB_LIBS)
+libnovell_la_LIBADD  = $(GLIB_LIBS) ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/oscar/Makefile.am pidgin-2.1.0/libpurple/protocols/oscar/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/oscar/Makefile.am	2007-06-12 09:13:03 +0300
+++ pidgin-2.1.0/libpurple/protocols/oscar/Makefile.am	2007-08-18 16:35:58 +0300
@@ -62,13 +62,13 @@
 st =
 pkg_LTLIBRARIES     = liboscar.la libaim.la libicq.la
 liboscar_la_SOURCES = $(OSCARSOURCES)
-liboscar_la_LIBADD  = $(GLIB_LIBS)
+liboscar_la_LIBADD  = $(GLIB_LIBS) ../../libpurple.la
 
 libaim_la_SOURCES = libaim.c
-libaim_la_LIBADD = liboscar.la
+libaim_la_LIBADD = liboscar.la ../../libpurple.la
 
 libicq_la_SOURCES = libicq.c
-libicq_la_LIBADD = liboscar.la
+libicq_la_LIBADD = liboscar.la ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/qq/Makefile.am pidgin-2.1.0/libpurple/protocols/qq/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/qq/Makefile.am	2007-06-23 06:06:32 +0300
+++ pidgin-2.1.0/libpurple/protocols/qq/Makefile.am	2007-08-18 16:40:12 +0300
@@ -85,7 +85,7 @@
 st =
 pkg_LTLIBRARIES  = libqq.la
 libqq_la_SOURCES = $(QQSOURCES)
-libqq_la_LIBADD  = $(GLIB_LIBS)
+libqq_la_LIBADD  = $(GLIB_LIBS) ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/simple/Makefile.am pidgin-2.1.0/libpurple/protocols/simple/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/simple/Makefile.am	2007-05-03 21:14:10 +0300
+++ pidgin-2.1.0/libpurple/protocols/simple/Makefile.am	2007-08-18 16:42:11 +0300
@@ -25,7 +25,7 @@
 st =
 pkg_LTLIBRARIES   = libsimple.la
 libsimple_la_SOURCES = $(SIMPLESOURCES)
-libsimple_la_LIBADD  = $(GLIB_LIBS)
+libsimple_la_LIBADD  = $(GLIB_LIBS) ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/yahoo/Makefile.am pidgin-2.1.0/libpurple/protocols/yahoo/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/yahoo/Makefile.am	2007-07-20 05:55:58 +0300
+++ pidgin-2.1.0/libpurple/protocols/yahoo/Makefile.am	2007-08-18 16:43:03 +0300
@@ -43,7 +43,7 @@
 st =
 pkg_LTLIBRARIES     = libyahoo.la
 libyahoo_la_SOURCES = $(YAHOOSOURCES)
-libyahoo_la_LIBADD  = $(GLIB_LIBS)
+libyahoo_la_LIBADD  = $(GLIB_LIBS) ../../libpurple.la
 
 endif
 
diff -urN pidgin-2.1.0.orig/libpurple/protocols/zephyr/Makefile.am pidgin-2.1.0/libpurple/protocols/zephyr/Makefile.am
--- pidgin-2.1.0.orig/libpurple/protocols/zephyr/Makefile.am	2007-05-03 21:14:10 +0300
+++ pidgin-2.1.0/libpurple/protocols/zephyr/Makefile.am	2007-08-18 16:44:14 +0300
@@ -81,7 +81,7 @@
 noinst_LIBRARIES = libzephyr.a
 libzephyr_a_SOURCES = $(ZEPHYRSOURCES)
 libzephyr_a_CFLAGS  = $(AM_CFLAGS)
-libzephyr_a_LIBADD  = $(ZEPHYRLIBS)
+libzephyr_a_LIBADD  = $(ZEPHYRLIBS) ../../libpurple.la
 
 else
 
@@ -90,10 +90,10 @@
 
 if EXTERNAL_LIBZEPHYR
 libzephyr_la_SOURCES = $(ZEPHYRSOURCESEXT)
-libzephyr_la_LIBADD = $(GLIB_LIBS) $(ZEPHYRLIBSEXT)
+libzephyr_la_LIBADD = $(GLIB_LIBS) $(ZEPHYRLIBSEXT) ../../libpurple.la
 else
 libzephyr_la_SOURCES = $(ZEPHYRSOURCES)
-libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBS)
+libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBS) ../../libpurple.la
 endif
 
 
