diff -ur pidgin-2.1.0.orig/finch/plugins/Makefile.am pidgin-2.1.0/finch/plugins/Makefile.am
--- pidgin-2.1.0.orig/finch/plugins/Makefile.am	2007-06-19 06:34:24 +0300
+++ pidgin-2.1.0/finch/plugins/Makefile.am	2007-08-18 18:13:37 +0300
@@ -21,10 +21,10 @@
 gntclipboard_la_CFLAGS = $(X11_CFLAGS)
 gntgf_la_CFLAGS = $(X11_CFLAGS)
 
-gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS)
-gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la
-gnthistory_la_LIBADD  = $(GLIB_LIBS)
-gntlastlog_la_LIBADD  = $(GLIB_LIBS)
+gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS) ../libgnt/libgnt.la ../../libpurple/libpurple.la
+gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la ../../libpurple/libpurple.la
+gnthistory_la_LIBADD  = $(GLIB_LIBS) ../../libpurple/libpurple.la
+gntlastlog_la_LIBADD  = $(GLIB_LIBS) ../../libpurple/libpurple.la ../libgnt/libgnt.la
 
 endif # PLUGINS
 
