diff -ur pidgin-2.5.3/finch/plugins/Makefile.am pidgin-2.5.3.new/finch/plugins/Makefile.am
--- pidgin-2.5.3/finch/plugins/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/finch/plugins/Makefile.am	2008-12-22 20:08:23 +0300
@@ -24,11 +24,11 @@
 gntclipboard_la_CFLAGS = $(X11_CFLAGS)
 gntgf_la_CFLAGS = $(X11_CFLAGS)
 
-gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS)
-gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la
-gnthistory_la_LIBADD  = $(GLIB_LIBS)
-gntlastlog_la_LIBADD  = $(GLIB_LIBS)
-grouping_la_LIBADD    = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la
+gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gnthistory_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+gntlastlog_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la $(top_builddir)/finch/libgnt/libgnt.la
+grouping_la_LIBADD    = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
 
 endif # PLUGINS
 
diff -ur pidgin-2.5.3/libpurple/Makefile.am pidgin-2.5.3.new/libpurple/Makefile.am
--- pidgin-2.5.3/libpurple/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/Makefile.am	2008-12-22 20:08:23 +0300
@@ -30,7 +30,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = purple.pc
 
-SUBDIRS = $(GCONF_DIR) plugins protocols tests . example
+SUBDIRS = . $(GCONF_DIR) plugins protocols tests example
 
 purple_coresources = \
 	account.c \
diff -ur pidgin-2.5.3/libpurple/plugins/Makefile.am pidgin-2.5.3.new/libpurple/plugins/Makefile.am
--- pidgin-2.5.3/libpurple/plugins/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/plugins/Makefile.am	2008-12-22 20:08:23 +0300
@@ -87,28 +87,28 @@
 simple_la_SOURCES			= simple.c
 statenotify_la_SOURCES      = statenotify.c
 
-autoaccept_la_LIBADD        = $(GLIB_LIBS)
-buddynote_la_LIBADD         = $(GLIB_LIBS)
-ciphertest_la_LIBADD		= $(GLIB_LIBS)
-codeinline_la_LIBADD		= $(GLIB_LIBS)
-idle_la_LIBADD              = $(GLIB_LIBS)
-joinpart_la_LIBADD          = $(GLIB_LIBS)
-log_reader_la_LIBADD        = $(GLIB_LIBS)
-newline_la_LIBADD           = $(GLIB_LIBS)
+autoaccept_la_LIBADD        = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+buddynote_la_LIBADD         = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+ciphertest_la_LIBADD		= $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+codeinline_la_LIBADD		= $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+idle_la_LIBADD              = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+joinpart_la_LIBADD          = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+log_reader_la_LIBADD        = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+newline_la_LIBADD           = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 notify_example_la_LIBADD    = $(GLIB_LIBS)
-offlinemsg_la_LIBADD        = $(GLIB_LIBS)
-pluginpref_example_la_LIBADD = $(GLIB_LIBS)
-psychic_la_LIBADD           = $(GLIB_LIBS)
-signals_test_la_LIBADD		= $(GLIB_LIBS)
-simple_la_LIBADD			= $(GLIB_LIBS)
-statenotify_la_LIBADD       = $(GLIB_LIBS)
+offlinemsg_la_LIBADD        = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+pluginpref_example_la_LIBADD = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+psychic_la_LIBADD           = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+signals_test_la_LIBADD		= $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+simple_la_LIBADD			= $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+statenotify_la_LIBADD       = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 if ENABLE_DBUS
 
 CLEANFILES              = dbus-example-bindings.c
 dbus_example_la_SOURCES = dbus-example.c
 
-dbus_example_la_LIBADD      = $(GLIB_LIBS) $(DBUS_LIBS)
+dbus_example_la_LIBADD      = $(GLIB_LIBS) $(DBUS_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 .PHONY: always
 
diff -ur pidgin-2.5.3/libpurple/plugins/ssl/Makefile.am pidgin-2.5.3.new/libpurple/plugins/ssl/Makefile.am
--- pidgin-2.5.3/libpurple/plugins/ssl/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/plugins/ssl/Makefile.am	2008-12-22 20:08:23 +0300
@@ -39,9 +39,9 @@
 ssl_gnutls_la_SOURCES = ssl-gnutls.c
 ssl_nss_la_SOURCES    = ssl-nss.c
 
-ssl_la_LIBADD        = $(GLIB_LIBS)
-ssl_gnutls_la_LIBADD = $(GLIB_LIBS) $(GNUTLS_LIBS)
-ssl_nss_la_LIBADD    = $(GLIB_LIBS) $(NSS_LIBS)
+ssl_la_LIBADD        = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+ssl_gnutls_la_LIBADD = $(GLIB_LIBS) $(GNUTLS_LIBS) $(top_builddir)/libpurple/libpurple.la
+ssl_nss_la_LIBADD    = $(GLIB_LIBS) $(NSS_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif # PLUGINS
 
Только в pidgin-2.5.3.new/libpurple/plugins/ssl: Makefile.am.orig
diff -ur pidgin-2.5.3/libpurple/protocols/bonjour/Makefile.am pidgin-2.5.3.new/libpurple/protocols/bonjour/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/bonjour/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/bonjour/Makefile.am	2008-12-22 20:08:23 +0300
@@ -40,7 +40,7 @@
 st =
 pkg_LTLIBRARIES       = libbonjour.la
 libbonjour_la_SOURCES = $(BONJOURSOURCES)
-libbonjour_la_LIBADD  = $(GLIB_LIBS) $(LIBXML_LIBS) $(AVAHI_LIBS)
+libbonjour_la_LIBADD  = $(GLIB_LIBS) $(LIBXML_LIBS) $(AVAHI_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/gg/Makefile.am pidgin-2.5.3.new/libpurple/protocols/gg/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/gg/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/gg/Makefile.am	2008-12-22 20:08:23 +0300
@@ -62,7 +62,7 @@
 st = $(GADU_CFLAGS)
 pkg_LTLIBRARIES = libgg.la
 libgg_la_SOURCES = $(GGSOURCES)
-libgg_la_LIBADD  = $(GLIB_LIBS) $(GADU_LIBS)
+libgg_la_LIBADD  = $(GLIB_LIBS) $(GADU_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/irc/Makefile.am pidgin-2.5.3.new/libpurple/protocols/irc/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/irc/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/irc/Makefile.am	2008-12-22 20:08:23 +0300
@@ -27,7 +27,7 @@
 st =
 pkg_LTLIBRARIES   = libirc.la
 libirc_la_SOURCES = $(IRCSOURCES)
-libirc_la_LIBADD  = $(GLIB_LIBS)
+libirc_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/jabber/Makefile.am pidgin-2.5.3.new/libpurple/protocols/jabber/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/jabber/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/jabber/Makefile.am	2008-12-22 20:16:29 +0300
@@ -68,10 +68,10 @@
 st =
 pkg_LTLIBRARIES      = libjabber.la libxmpp.la
 libjabber_la_SOURCES = $(JABBERSOURCES)
-libjabber_la_LIBADD  = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS)
+libjabber_la_LIBADD  = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 libxmpp_la_SOURCES = libxmpp.c
-libxmpp_la_LIBADD = libjabber.la
+libxmpp_la_LIBADD = libjabber.la $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/msn/Makefile.am pidgin-2.5.3.new/libpurple/protocols/msn/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/msn/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/msn/Makefile.am	2008-12-22 20:08:23 +0300
@@ -83,7 +83,7 @@
 st =
 pkg_LTLIBRARIES   = libmsn.la
 libmsn_la_SOURCES = $(MSNSOURCES)
-libmsn_la_LIBADD  = $(GLIB_LIBS)
+libmsn_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/msnp9/Makefile.am pidgin-2.5.3.new/libpurple/protocols/msnp9/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/msnp9/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/msnp9/Makefile.am	2008-12-22 20:08:23 +0300
@@ -79,7 +79,7 @@
 st =
 pkg_LTLIBRARIES   = libmsn.la
 libmsn_la_SOURCES = $(MSNP9SOURCES)
-libmsn_la_LIBADD  = $(GLIB_LIBS)
+libmsn_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/myspace/Makefile.am pidgin-2.5.3.new/libpurple/protocols/myspace/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/myspace/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/myspace/Makefile.am	2008-12-22 20:08:23 +0300
@@ -33,7 +33,7 @@
 st =
 pkg_LTLIBRARIES       = libmyspace.la
 libmyspace_la_SOURCES = $(MSIMSOURCES)
-libmyspace_la_LIBADD  = $(GLIB_LIBS)
+libmyspace_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/novell/Makefile.am pidgin-2.5.3.new/libpurple/protocols/novell/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/novell/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/novell/Makefile.am	2008-12-22 20:08:23 +0300
@@ -42,7 +42,7 @@
 st =
 pkg_LTLIBRARIES      = libnovell.la
 libnovell_la_SOURCES = $(NOVELLSOURCES)
-libnovell_la_LIBADD  = $(GLIB_LIBS)
+libnovell_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/oscar/Makefile.am pidgin-2.5.3.new/libpurple/protocols/oscar/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/oscar/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/oscar/Makefile.am	2008-12-22 20:17:53 +0300
@@ -61,13 +61,13 @@
 st =
 pkg_LTLIBRARIES     = liboscar.la libaim.la libicq.la
 liboscar_la_SOURCES = $(OSCARSOURCES)
-liboscar_la_LIBADD  = $(GLIB_LIBS)
+liboscar_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 libaim_la_SOURCES   = libaim.c
-libaim_la_LIBADD    = liboscar.la
+libaim_la_LIBADD    = liboscar.la $(top_builddir)/libpurple/libpurple.la
 
 libicq_la_SOURCES   = libicq.c
-libicq_la_LIBADD    = liboscar.la
+libicq_la_LIBADD    = liboscar.la $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/qq/Makefile.am pidgin-2.5.3.new/libpurple/protocols/qq/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/qq/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/qq/Makefile.am	2008-12-22 20:08:23 +0300
@@ -65,7 +65,7 @@
 st =
 pkg_LTLIBRARIES  = libqq.la
 libqq_la_SOURCES = $(QQSOURCES)
-libqq_la_LIBADD  = $(GLIB_LIBS)
+libqq_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/sametime/Makefile.am pidgin-2.5.3.new/libpurple/protocols/sametime/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/sametime/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/sametime/Makefile.am	2008-12-22 20:21:45 +0300
@@ -27,7 +27,7 @@
 endif
 
 libsametime_la_LDFLAGS = -module -avoid-version
-libsametime_la_LIBADD = $(GLIB_LIBS) $(MEANWHILE_LIBS)
+libsametime_la_LIBADD = $(GLIB_LIBS) $(MEANWHILE_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/libpurple \
diff -ur pidgin-2.5.3/libpurple/protocols/silc/Makefile.am pidgin-2.5.3.new/libpurple/protocols/silc/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/silc/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/silc/Makefile.am	2008-12-22 20:22:17 +0300
@@ -34,7 +34,7 @@
 st = $(SILC_CFLAGS)
 pkg_LTLIBRARIES          = libsilcpurple.la
 libsilcpurple_la_SOURCES = $(SILCSOURCES)
-libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS)
+libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/silc10/Makefile.am pidgin-2.5.3.new/libpurple/protocols/silc10/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/silc10/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/silc10/Makefile.am	2008-12-22 20:22:33 +0300
@@ -24,7 +24,7 @@
 st = $(SILC_CFLAGS)
 pkg_LTLIBRARIES          = libsilcpurple.la
 libsilcpurple_la_SOURCES = $(SILCSOURCES)
-libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS)
+libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/simple/Makefile.am pidgin-2.5.3.new/libpurple/protocols/simple/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/simple/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/simple/Makefile.am	2008-12-22 20:08:23 +0300
@@ -25,7 +25,7 @@
 st =
 pkg_LTLIBRARIES      = libsimple.la
 libsimple_la_SOURCES = $(SIMPLESOURCES)
-libsimple_la_LIBADD  = $(GLIB_LIBS)
+libsimple_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/yahoo/Makefile.am pidgin-2.5.3.new/libpurple/protocols/yahoo/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/yahoo/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/yahoo/Makefile.am	2008-12-22 20:08:23 +0300
@@ -45,7 +45,7 @@
 st =
 pkg_LTLIBRARIES     = libyahoo.la
 libyahoo_la_SOURCES = $(YAHOOSOURCES)
-libyahoo_la_LIBADD  = $(GLIB_LIBS)
+libyahoo_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/libpurple/protocols/zephyr/Makefile.am pidgin-2.5.3.new/libpurple/protocols/zephyr/Makefile.am
--- pidgin-2.5.3/libpurple/protocols/zephyr/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/libpurple/protocols/zephyr/Makefile.am	2008-12-22 20:19:47 +0300
@@ -89,10 +89,10 @@
 
 if EXTERNAL_LIBZEPHYR
 libzephyr_la_SOURCES = $(ZEPHYRSOURCESEXT)
-libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBSEXT)
+libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBSEXT) $(top_builddir)/libpurple/libpurple.la
 else
 libzephyr_la_SOURCES = $(ZEPHYRSOURCES)
-libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBS)
+libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBS) $(top_builddir)/libpurple/libpurple.la
 endif
 
 
diff -ur pidgin-2.5.3/pidgin/plugins/cap/Makefile.am pidgin-2.5.3.new/pidgin/plugins/cap/Makefile.am
--- pidgin-2.5.3/pidgin/plugins/cap/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/pidgin/plugins/cap/Makefile.am	2008-12-22 20:08:23 +0300
@@ -15,7 +15,7 @@
 
 endif
 
-cap_la_LIBADD = $(GTK_LIBS) $(SQLITE3_LIBS)
+cap_la_LIBADD = $(GTK_LIBS) $(SQLITE3_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 AM_CPPFLAGS = \
 	-DDATADIR=\"$(datadir)\" \
diff -ur pidgin-2.5.3/pidgin/plugins/gestures/Makefile.am pidgin-2.5.3.new/pidgin/plugins/gestures/Makefile.am
--- pidgin-2.5.3/pidgin/plugins/gestures/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/pidgin/plugins/gestures/Makefile.am	2008-12-22 20:08:23 +0300
@@ -13,7 +13,7 @@
 	stroke.c \
 	stroke-draw.c
 
-gestures_la_LIBADD = $(GTK_LIBS)
+gestures_la_LIBADD = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/pidgin/plugins/gevolution/Makefile.am pidgin-2.5.3.new/pidgin/plugins/gevolution/Makefile.am
--- pidgin-2.5.3/pidgin/plugins/gevolution/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/pidgin/plugins/gevolution/Makefile.am	2008-12-22 20:08:23 +0300
@@ -15,7 +15,7 @@
 	new_person_dialog.c \
 	eds-utils.c
 
-gevolution_la_LIBADD = $(EVOLUTION_ADDRESSBOOK_LIBS) $(GTK_LIBS)
+gevolution_la_LIBADD = $(EVOLUTION_ADDRESSBOOK_LIBS) $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
diff -ur pidgin-2.5.3/pidgin/plugins/Makefile.am pidgin-2.5.3.new/pidgin/plugins/Makefile.am
--- pidgin-2.5.3/pidgin/plugins/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/pidgin/plugins/Makefile.am	2008-12-22 20:08:23 +0300
@@ -86,22 +86,22 @@
 timestamp_format_la_SOURCES = timestamp_format.c
 xmppconsole_la_SOURCES      = xmppconsole.c
 
-convcolors_la_LIBADD        = $(GTK_LIBS)
-contact_priority_la_LIBADD  = $(GTK_LIBS)
-extplacement_la_LIBADD      = $(GTK_LIBS)
-gtk_signals_test_la_LIBADD  = $(GTK_LIBS)
-gtkbuddynote_la_LIBADD      = $(GTK_LIBS)
-history_la_LIBADD           = $(GTK_LIBS)
-iconaway_la_LIBADD          = $(GTK_LIBS)
-markerline_la_LIBADD        = $(GTK_LIBS)
-notify_la_LIBADD            = $(GTK_LIBS)
-pidginrc_la_LIBADD          = $(GTK_LIBS)
-relnot_la_LIBADD            = $(GLIB_LIBS)
-sendbutton_la_LIBADD        = $(GTK_LIBS)
-spellchk_la_LIBADD          = $(GTK_LIBS)
-timestamp_la_LIBADD         = $(GTK_LIBS)
-timestamp_format_la_LIBADD  = $(GTK_LIBS)
-xmppconsole_la_LIBADD       = $(GTK_LIBS)
+convcolors_la_LIBADD        = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+contact_priority_la_LIBADD  = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+extplacement_la_LIBADD      = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+gtk_signals_test_la_LIBADD  = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+gtkbuddynote_la_LIBADD      = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+history_la_LIBADD           = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+iconaway_la_LIBADD          = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+markerline_la_LIBADD        = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+notify_la_LIBADD            = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+pidginrc_la_LIBADD          = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+relnot_la_LIBADD            = $(GLIB_LIBS) $(top_builddir)/libpurple/libpurple.la
+sendbutton_la_LIBADD        = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+spellchk_la_LIBADD          = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+timestamp_la_LIBADD         = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+timestamp_format_la_LIBADD  = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
+xmppconsole_la_LIBADD       = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif # PLUGINS
 
diff -ur pidgin-2.5.3/pidgin/plugins/musicmessaging/Makefile.am pidgin-2.5.3.new/pidgin/plugins/musicmessaging/Makefile.am
--- pidgin-2.5.3/pidgin/plugins/musicmessaging/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/pidgin/plugins/musicmessaging/Makefile.am	2008-12-22 20:08:23 +0300
@@ -16,7 +16,7 @@
 musicmessaging_la_SOURCES = \
 	musicmessaging.c
 
-musicmessaging_la_LIBADD = $(GTK_LIBS) $(DBUS_LIBS)
+musicmessaging_la_LIBADD = $(GTK_LIBS) $(DBUS_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 CLEANFILES              = music-messaging-bindings.c
 
diff -ur pidgin-2.5.3/pidgin/plugins/ticker/Makefile.am pidgin-2.5.3.new/pidgin/plugins/ticker/Makefile.am
--- pidgin-2.5.3/pidgin/plugins/ticker/Makefile.am	2008-12-21 03:38:16 +0300
+++ pidgin-2.5.3.new/pidgin/plugins/ticker/Makefile.am	2008-12-22 20:08:23 +0300
@@ -14,7 +14,7 @@
 	gtkticker.h \
 	ticker.c
 
-ticker_la_LIBADD = $(GTK_LIBS)
+ticker_la_LIBADD = $(GTK_LIBS) $(top_builddir)/libpurple/libpurple.la
 
 endif
 
