diff -ur pidgin-2.1.0.orig/libpurple/Makefile.am pidgin-2.1.0/libpurple/Makefile.am
--- pidgin-2.1.0.orig/libpurple/Makefile.am	2007-07-21 22:28:01 +0300
+++ pidgin-2.1.0/libpurple/Makefile.am	2007-08-18 17:12:11 +0300
@@ -29,7 +29,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = purple.pc
 
-SUBDIRS = $(GCONF_DIR) plugins protocols tests . example
+SUBDIRS = . $(GCONF_DIR) plugins protocols tests example
 
 purple_coresources = \
 	account.c \
