--- pidgin-2.6.1/libpurple/proxy.c.alt	2009-08-25 14:00:00 +0400
+++ pidgin-2.6.1/libpurple/proxy.c	2009-08-25 14:00:37 +0400
@@ -245,7 +245,7 @@ purple_gnome_proxy_get_info(void)
 		return &info;
 	}
 
-	if (purple_strequal(tmp, "manual\n")) {
+	if (!purple_strequal(tmp, "manual\n")) {
 		/* Unknown setting.  Fallback to using our global proxy settings. */
 		g_free(tmp);
 		return purple_global_proxy_get_info();
--- pidgin-2.6.1/pidgin/gtkprefs.c.alt	2009-08-19 12:12:48 +0400
+++ pidgin-2.6.1/pidgin/gtkprefs.c	2009-08-25 13:59:38 +0400
@@ -1724,7 +1724,7 @@ proxy_button_clicked_cb(GtkWidget *butto
 {
 	GError *err = NULL;
 
-	if (g_spawn_command_line_async ("gnome-network-preferences", &err))
+	if (g_spawn_command_line_async ("gnome-network-properties", &err))
 		return;
 
 	purple_notify_error(NULL, NULL, _("Cannot start proxy configuration program."), err->message);
@@ -2040,7 +2040,7 @@ network_page(void)
 		gtk_widget_hide(table);
 	} else if (purple_running_gnome()) {
 		gchar *path;
-		path = g_find_program_in_path("gnome-network-preferences");
+		path = g_find_program_in_path("gnome-network-properties");
 		if (path != NULL) {
 			gtk_widget_set_sensitive(proxy_button, TRUE);
 			gtk_widget_hide(proxy_warning);
